### PR TITLE
welcome: Add note about maildir specifics

### PR DESCRIPTION
**Pull request by vmo69**
DOV-5757

### CODE REVIEW

File: source/configuration_manual/welcome_plugin.rst [hunk 1]
```diff
@@ -10,6 +10,12 @@ Call a script when the user logs in for the first time. This is specifically
 done when the INBOX is (auto)created. The scripts are called similarly to
 :ref:`quota_configuration_warning_scripts`.
 
+.. note:: The welcome plugin relies on creation or autocreation of INBOX.
+          If the mail_location is maildir there can be situations where the
+          welcome script is not triggered. That is the case when the user
+          home dir is created by a IMAP command that does not interact with
```
Comment (sirainen) [inline source/configuration_manual/welcome_plugin.rst:16]: Isn't this fixed for Maildir by your changes?

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/configuration_manual/welcome_plugin.rst (hunk 1 above)
Comment (sirainen) [reply source/configuration_manual/welcome_plugin.rst:16]: Maybe this could be a .. changedsince, and say this was fixed for everything except Maildir?

Comment (sirainen) [reply source/configuration_manual/welcome_plugin.rst:16]: And I suppose it's Maildir only when INBOX = Maildir/ directory, so e.g. using mail_location = maildir:~/Maildir:INBOX=~/Maildir/.INBOX works around this?

Comment (vmo69) [reply source/configuration_manual/welcome_plugin.rst]: Yes right, only happens without the non-default INBOX location. I added that to the description here.

File: source/configuration_manual/welcome_plugin.rst [hunk 2]
```diff
@@ -10,6 +10,17 @@ Call a script when the user logs in for the first time. This is specifically
...
 :ref:`quota_configuration_warning_scripts`.
 
+.. versionchanged:: v2.4;v3.0 The welcome plugin relies on creation or autocreation
+                  of INBOX. If the mail_location is maildir and the INBOX location
+                  is left default, there can be situations where the welcome script
+                  is not triggered. That is the case when the user home dir is
+                  created by an IMAP command that does not interact with INBOX
+                  directly for example GETMETADATA.
+
+                  Use maildir:~/Maildir:INBOX=~/Maildir/.INBOX to prevent this
```
Comment (sirainen) [inline source/configuration_manual/welcome_plugin.rst:20]: Add `` quotes to `maildir:~/Maildir:INBOX=~/Maildir/.INBOX`

**Review by sirainen — CHANGES_REQUESTED** [review]

**Pull request closed without merging**